src/main/java/uk/gov/ch/exception/EntityEmailNotFoundException.java

```diff
@@ -0,0 +1,11 @@
+package uk.gov.ch.exception;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus(HttpStatus.NOT_FOUND)
+public class EntityEmailNotFoundException extends RuntimeException {
```

Comment:
I would get 'Overseas' into the name of this class to make it more specific.

I would also prefer that 'email address' rather than just 'email' was referenced as they are two different things (although up to you, as I know that just 'email' is used in lots of places in CH code). But... you do use `emailAddress` in the Data POJO, so consistency should also be considered.

So `OverseasEntityEmailAddressNotFoundException`.